### PR TITLE
Replace Layer Names with Layer enum

### PR DESCRIPTION
**Pull request by daleeidd**
Deprecates the _Layer Name_ and adds _Layer_ to supersede it. Since a lot of people are going to hit this, I put a fair bit of effort in making sure the transition is as smooth as possible. This should make things easier in the future.

layer2

I have groups changes into commits for easier digestion. Notable changes are:
- Predicated now works with strings
- Add call-to-action (eg "Click fix button") to help box message
- Add Layer attribute
- Add HelpBox attribute (it uses predicated to show/hide itself)
- Deprecated layer name. It will still be used if data is there
- Add Layer. Cannot be used until migrated

The plan is that in some future version we just remove the Layer Name and ignore the old data. By that time hopefully enough people have migrated.

I have left our scenes alone for testing/reviewing. I will migrate them once reviewed.

TODO
- Migrate our own scenes
- Test on 2020/2021

### CODE REVIEW

**Review by huwb — APPROVED** [review]
Maybe it doesnt matter but the layer name is hidden, so the user wont know what layer they had chosen. However i guess 99% of hte time it will be 'water'

It occurs to me we could probably migrate the data automatically - if a valid layer name is found, set it the layer field, clear the layer name, make the OceanRenderer component dirty. not critical for this but just a thought, and im probably going to look at auto migrating the spectrum data after the latest round of changes merge, so its something thats on my mind

I'll let you decide how to proceed, merging what you have works for me, so ill approve

File: crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs
Comment (daleeidd) [reply crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs]: I just checked and most of the obsolete copy is redundant. I have rewrote it to be concise. They now appear in the console as:

```log
Assets/Crest/Crest/Scripts/OceanRenderer.cs(435,23): warning CS0618: 'OceanRenderer._layerName' is obsolete: 'Use the _layer field instead.'
```

```log
Assets/Crest/Crest/Scripts/OceanRenderer.cs(436,23): warning CS0618: 'OceanRenderer.LayerName' is obsolete: 'Use the Layer property instead.'
```

File: crest/Assets/Development/Editor/AttributeValidator.cs
Comment (daleeidd) [reply crest/Assets/Development/Editor/AttributeValidator.cs:29]: It won't.

**Comment by daleeidd** [comment]
> Maybe it doesnt matter but the layer name is hidden, so the user wont know what layer they had chosen. However i guess 99% of hte time it will be 'water'

I have added the layer name as part of the fix description. I could create a new attribute or do something with Predicated which hides properties dynamically.
 
> It occurs to me we could probably migrate the data automatically - if a valid layer name is found, set it the layer field, clear the layer name, make the OceanRenderer component dirty. not critical for this but just a thought, and im probably going to look at auto migrating the spectrum data after the latest round of changes merge, so its something thats on my mind

Hmmm not a bad idea. It would have to be clear that something has changed like logging to the console if we mark it as dirty. I will go with this route for this one and see how it goes.

**Pull request merged**
